crates/rust-analyzer/src/reload.rs

```diff
@@ -629,6 +629,10 @@ pub(crate) fn should_refresh_for_change(path: &AbsPath, change_kind: ChangeKind)
     if IMPLICIT_TARGET_DIRS.iter().any(|it| parent.as_ref().ends_with(it)) {
         return true;
     }
+    if (file_name == "config.toml" || file_name == "config") && parent.as_ref().ends_with(".cargo")
```

Comment:
I don't think this will ever get triggered because there's an early-return above when the file extension is anything but `.rs`

Reply:
You're right. Should be resolved now.